blocks/library/embed/index.js

```diff
@@ -118,17 +144,24 @@ registerBlockType( 'core/embed', {
 						} else {
 							this.setState( { error: true } );
 						}
-						this.setState( { fetching: false } );
+						this.setState( { fetching: false, loadingFromSavedBlock: false } );
 					} );
 				}
 			);
 		}
 
 		render() {
-			const { html, type, error, fetching } = this.state;
+			const { html, type, error, fetching, loadingFromSavedBlock } = this.state;
```

Comment:
I think ideally this should be the single `loading` state, and it doesn't matter whether you just input the url in the embed field, or if it's loading from saved content.